Source/LibationWinForms/grid/FilterableSortableBindingList.cs

```diff
@@ -0,0 +1,99 @@
+﻿using ApplicationServices;
+using Dinah.Core.DataBinding;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace LibationWinForms
+{
+	/*
+     * Allows filtering of the underlying SortableBindingList<GridEntry>
+     * by implementing IBindingListView and using SearchEngineCommands 
+     * 
+     * When filtering is applied, the filtered-out items are removed
+     * from the base list and added to the private FilterRemoved list.
+     * When filtering is removed, items in the FilterRemoved list are
+     * added back to the base list.
+     * 
+     * Remove is overridden to ensure that removed items are removed from
+     * the base list (visible items) as well as the FilterRemoved list.
+     */
+	internal class FilterableSortableBindingList : SortableBindingList<GridEntry>, IBindingListView
+	{
+		/// <summary>
+		/// Items that were removed from the base list due to filtering
+		/// </summary>
+		private readonly List<GridEntry> FilterRemoved = new();
+		private string FilterString;
+		private Action Sort;
+		public FilterableSortableBindingList(IEnumerable<GridEntry> enumeration) : base(enumeration)
+		{
+			//This is only necessary because SortableBindingList doesn't expose Sort()
+			//You should make SortableBindingList.Sort protected and remove reflection
+			var method = typeof(SortableBindingList<GridEntry>).GetMethod("Sort", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
```

Comment:
Pushing that update now. Nuget builds are slow this time of day

Reply:
Push complete. Merge to see it

Reply:
Christ I'm slow today.  I didn't understand I needed to merge your changes. Done now.